airflow/serialization/serialized_objects.py

```diff
@@ -438,7 +438,7 @@ def serialize(
             json_pod = PodGenerator.serialize_pod(var)
             return cls._encode(json_pod, type_=DAT.POD)
         elif isinstance(var, DAG):
-            return SerializedDAG.serialize_dag(var)
+            return cls._encode(SerializedDAG.serialize_dag(var), type_=DAT.DAG)
```

Comment:
There is no need to encode this, we expect it to have a top level "dag" unique key as shown in 

https://github.com/apache/airflow/blob/bf2d411a2a953d17f7847d23f0563589a0a58a94/tests/serialization/test_dag_serialization.py#L123


Similarly, we don't encode tasks or each operator, because we know the structure that it would be within `["dag"]["tasks"]`

Check L445 and L447

Reply:
When building DAG s10n it was designed to using Airflow internal knowledge to not inflate the final blob and optimize wherever possible. This is the reason we don't store defaults and objects that are None for DAG and operator objects

Reply:
Also, be careful about backwards-incompatible changes for this.

And if for AIP-44 we absolutely need this sort of change, that will probably be more involved than just the DAG object

Reply:
Right, but:
if I  serialize dag (with BaseSerialization.serialize) then try to deserialize it with BaseSerialization.deserialize it fails - as it doesn't get there
https://github.com/apache/airflow/blob/main/airflow/serialization/serialized_objects.py#L551
because 
```
        var = encoded_var[Encoding.VAR]
        type_ = encoded_var[Encoding.TYPE]
```
failes (key not exists).

- also we may need to send the DAG as part of bigger structure (e.g. https://github.com/apache/airflow/blob/main/airflow/dag_processing/processor.py#L882)


In code I am not able to find any usages of BaseSerialization.serialize on DAG (at least when looking for something like `BaseSerialization.serialize.*dag ` 

We are currently working on making sure all other objects types are serializable.

Reply:
Yeah. I think @mhenc is right - this part of code seems to be not used before - it's not 'DAG` serialization that gets afffected here, it's just serializing the whole `dag` s part of bigger structure. I have not seen any place in the code where we'd do that before.